### PR TITLE
Fix sending BC.ActivateApp twice

### PR DESCRIPTION
**Pull request by AByzhynar**
Separated handling of specific device app from handling for all devices

Related: [APPLINK-30842](https://adc.luxoft.com/jira/browse/APPLINK-30842)

### CODE REVIEW

**Comment by AByzhynar** [comment]
@Kozoriz @VProdanov @dev-gh @LuxoftAKutsan , please review, friends

**Comment by AByzhynar** [comment]
Error in commit message text will be corrected.

File: src/components/application_manager/src/policies/policy_handler.cc [hunk 1]
```diff
@@ -974,55 +974,59 @@ void PolicyHandler::OnAllowSDLFunctionalityNotification(
...
-    const std::string device_id = *it_ids;
+    std::vector<std::string>::const_iterator it_ids = device_macs.begin();
+    std::vector<std::string>::const_iterator it_ids_end = device_macs.end();
+    for (; it_ids != it_ids_end; ++it_ids) {
+      const std::string device_id = *it_ids;
+
+      if (kDefaultDeviceMacAddress == device_id) {
+        LOG4CXX_WARN(logger_,
+                     "Device with id " << device_id << " wasn't found.");
+        return;
```
Comment (dev-gh) [inline src/components/application_manager/src/policies/policy_handler.cc]: @AByzhynar 
I think 'continue' must be instead

File: src/components/application_manager/src/policies/policy_handler.cc [hunk 2]
```diff
@@ -974,55 +974,59 @@ void PolicyHandler::OnAllowSDLFunctionalityNotification(
...
+      }
+      policy_manager_->SetUserConsentForDevice(device_id, is_allowed);
+      uint32_t device_handle = 0;
+      if (!connection_handler.GetDeviceID(device_id, &device_handle)) {
+        LOG4CXX_WARN(logger_,
+                     "Device hadle with mac " << device_id << " wasn't found.");
```
Comment (dev-gh) [inline src/components/application_manager/src/policies/policy_handler.cc]: @AByzhynar 
typo: handle

File: src/components/application_manager/src/policies/policy_handler.cc [hunk 3]
```diff
@@ -1037,19 +1041,21 @@ void PolicyHandler::OnAllowSDLFunctionalityNotification(
...
     if (!app) {
       LOG4CXX_WARN(logger_,
                    "Application with id '"
                        << last_activated_app_id_
-                       << "' not found within registered applications.");
+                       << "' not found among registered applications.");
       return;
     }
-    if (app) {
+    if (app && is_allowed) {
```
Comment (dev-gh) [inline src/components/application_manager/src/policies/policy_handler.cc]: @AByzhynar 
Duplicated app check

File: src/components/application_manager/src/policies/policy_handler.cc [hunk 4]
```diff
@@ -1059,6 +1065,12 @@ void PolicyHandler::OnAllowSDLFunctionalityNotification(
       application_manager_.state_controller().SetRegularState(
           app, mobile_apis::HMILevel::HMI_FULL, state, true);
       last_activated_app_id_ = 0;
+    } else {
+      SDLAlowedNotification allow_notification(
```
Comment (dev-gh) [inline src/components/application_manager/src/policies/policy_handler.cc]: @AByzhynar 
Why in this case (device disallowed) you don't use DeactivateApplication as above?

File: src/components/application_manager/src/policies/policy_handler.cc (hunk 1 above)
Comment (AByzhynar) [inline src/components/application_manager/src/policies/policy_handler.cc]: @dev-gh Fixed in bb52aca0f6f680894d34fb8e11a116ef0cf57861

File: src/components/application_manager/src/policies/policy_handler.cc (hunk 2 above)
Comment (AByzhynar) [inline src/components/application_manager/src/policies/policy_handler.cc]: @dev-gh Fixed in bb52aca0f6f680894d34fb8e11a116ef0cf57861

File: src/components/application_manager/src/policies/policy_handler.cc (hunk 3 above)
Comment (AByzhynar) [inline src/components/application_manager/src/policies/policy_handler.cc]: @dev-gh Fixed in bb52aca0f6f680894d34fb8e11a116ef0cf57861

File: src/components/application_manager/src/policies/policy_handler.cc (hunk 4 above)
Comment (AByzhynar) [inline src/components/application_manager/src/policies/policy_handler.cc]: @dev-gh Fixed in bb52aca0f6f680894d34fb8e11a116ef0cf57861

**Comment by AByzhynar** [comment]
Rebuild required

**Comment by AByzhynar** [comment]
@LuxoftAKutsan Please merge this PR

**Pull request merged**